lalrpop/src/build/mod.rs

```diff
@@ -87,6 +87,24 @@ fn gen_resolve_file(session: &Session, lalrpop_file: &Path, ext: &str) -> io::Re
         in_dir
     };
 
+    // Ideally we do something like syn::parse_str::<syn::Ident>(lalrpop_file.file_name())?;
+    // But I don't think we want a full blown syn dependency unless fully converting to proc macros.
+    if lalrpop_file
+        .file_name()
+        .unwrap()
+        .to_str()
+        .unwrap()
```

Comment:
This unwrap panics if the filename contains invalid unicode.  I feel like adding a check for one sort of weird filename that introduces a panic on a different sort feels weird to me.  Maybe `unwrap_or_else("")` to skip the check if the conversion fails would be cleaner?

Reply:
Hmm, I certainly could but maybe it's not a bad thing that it will panic on invalid unicode? I don't think there is an expectation that any of this would work in that case... what name would the module have that the rust code would call? How would you supply the path to lalrpop_mod if it isn't a valid string?

Reply:
I tend to take the angle that if the failure is purely our internal bug, panicing is fine.  If it could possibly be triggered by external state, we should error instead.  So in my opinion erroring on invalid unicode instead of panicing would be better.

Reply:
To elaborate, maybe someone's build script wants to log and error and proceed to other tasks if lalrpop has an issue.  Returning an error instead of panicing gives them that option.

Reply:
I've replaced both of the panics with io Errors